packages/peregrine/lib/talons/CheckoutPage/PaymentInformation/useCreditCard.js

```diff
@@ -380,6 +394,32 @@ export const useCreditCard = props => {
      * Effects
      */
 
+    const hasNonce = !!(paymentNonceData && paymentNonceData.cart.paymentNonce);
+    const selectedMethodIsBraintree =
+        paymentMethodData &&
+        paymentMethodData.cart.selected_payment_method.code === 'braintree';
+
+    // On first render we should wait to know if we have previously saved data.
+    // If so, we can just tell the parent component to display the summary.
+    useEffect(() => {
+        if (paymentMethodData && paymentNonceData) {
+            if (isFirstMount && selectedMethodIsBraintree && hasNonce) {
```

Comment:
Per team review discussion I am `isFirstMount` and `shouldRender` to be more explicit. Also I might have to update this effect as I don't think it's quite correct.